### PR TITLE
Task/outbox

### PR DESCRIPTION
**Pull request by YuriyTretyakov**

### CODE REVIEW

File: BeerDispancer/BeerDispencer.Application/Abstractions/IDispencerUof.cs [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-using System.Transactions;
-using BeerDispancer.Application.DTO;
+﻿using System.Transactions;
 using BeerDispencer.Application.Abstractions;
-using BeerDispencer.Application.DTO;
 
 
 namespace BeerDispancer.Application.Abstractions
 {
-	public interface IDispencerUof : IDisposable
```
Comment (Dessyk) [inline BeerDispancer/BeerDispencer.Application/Abstractions/IDispencerUof.cs:10]: UOW -> Unit Of Work :-)

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs [hunk 2]
```diff
@@ -65,6 +93,8 @@ public void Dispose()
             _transaction?.Dispose();
```
Comment (Dessyk) [inline BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs:93]: Why not in dispose method?

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs [hunk 3]
```diff
@@ -33,28 +41,48 @@ public async Task Complete()
...
-        //   );
-        //}
+        
 
         public TransactionScope StartTransaction()
         {
             var transactionOptions = new TransactionOptions
             {
-                IsolationLevel = IsolationLevel.ReadCommitted,
+                IsolationLevel = IsolationLevel.ReadUncommitted,
```
Comment (Dessyk) [inline BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs]: Why?

**Review by Dessyk — COMMENTED** [review]
Where is the outbox events dispatching?

File: BeerDispancer/BeerDispencer.Application/Abstractions/IDispencerUof.cs (hunk 1 above)
Comment (YuriyTretyakov) [reply BeerDispancer/BeerDispencer.Application/Abstractions/IDispencerUof.cs:10]: IDK (

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs (hunk 3 above)
Comment (YuriyTretyakov) [reply BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs]: mess with different branches. corrected

**Comment by YuriyTretyakov** [comment]
> Where is the outbox events dispatching?
here
BeerDispancer/BeerDispencer.Infrastructure/Payment/PaymentProcessorJob.cs

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs (hunk 2 above)
Comment (YuriyTretyakov) [reply BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs:93]: code is looking like that: separate methods for COmmit and Dispose

  public void CommitTransaction()
        {
            _transaction.Complete();
        }

        public void Dispose()
        {
            _transaction?.Dispose();
            _dbcontext?.Dispose();
        }

File: BeerDispancer/BeerDispencer.Infrastructure/Payment/PaymentProcessorJob.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+namespace BeerDispencer.Infrastructure.Payment
+{
+    public class PaymentProcessorJob: BackgroundService
+    {
+       
+        private readonly IServiceScopeFactory facrory;
+        private readonly ILogger<PaymentProcessorJob> _logger;
+
+        public PaymentProcessorJob(IServiceScopeFactory facrory , ILogger<PaymentProcessorJob> logger, IHostApplicationLifetime applicationLifetime)
```
Comment (Dessyk) [inline BeerDispancer/BeerDispencer.Infrastructure/Payment/PaymentProcessorJob.cs:15]: How can you be sure that PaymentProcessorJob won't be run in 2 separate services?

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUOW.cs [hunk 5]
```diff
@@ -55,6 +54,35 @@ public TransactionScope StartTransaction()
...
+
+            await _dbcontext.Payments.AddAsync(payment);
+
+            await _dbcontext.Outbox.AddAsync(outbox);
+            await _dbcontext.SaveChangesAsync(_cancellationToken);
+            CommitTransaction();
+        }
+
         public void CommitTransaction()
         {
```
Comment (Dessyk) [inline BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUOW.cs:87]: Why not to dispatch here?

Comment (Dessyk) [reply BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUOW.cs:87]: Check Domain events pattern -> https://microservices.io/patterns/data/domain-event.html

File: BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs (hunk 2 above)
Comment (Dessyk) [reply BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUof.cs:93]: Mixed up) Sorry)

**Pull request closed without merging**